src/tabs/index.tsx

```diff
@@ -62,88 +72,125 @@ const generateTabPanelId = (tab: Tab):string => {
-        return (
-            <div className='kirk-tabs'>
-                <div role='tablist'
-                     aria-orientation='horizontal'
-                     aria-multiselectable='false'
-                     className='kirk-tablist'>
-                    {tabs.map((tab) => {
-                        const isSelected = selectedTab === tab
-                        return <button
-                            role='tab'
-                            aria-controls={`${generateTabPanelId(tab)}`}
-                            aria-selected={isSelected ? 'true' : 'false'}
-                            title={tab.label}
-                            tabIndex={isSelected ? 0 : -1}
-                            id={tab.id}
-                            key={tab.id}
-                            ref={this.state.tabIdToRefs.get(tab.id)}
-                            onClick={this.handleTabClicked}
-                            onKeyDown={this.handleTabKeyDown}
-                            className={cc(['kirk-tab', {'kirk-tab-selected': isSelected}])}>
-                            {tab.label}
-                        </button>
-                    })}
-                </div>
+    if (tabs.length === 0) {
+      return null
+    }
 
-                <div role='tabpanel' className='kirk-tab-panel'
-                     id={`${generateTabPanelId(selectedTab)}`}>
-                    {selectedTab.panelContent}
+    const { activeTabId } = this.state
+    const selectedTab = tabs.find(tab => activeTabId === tab.id)
+    const isFixedTabs = this.props.status === TabStatus.FIXED
+
+    return (
+      <div className={cc(['kirk-tabs', { 'kirk-tabs-fixed': isFixedTabs }])}>
+        <div
+          id="tablist"
+          role="tablist"
+          aria-orientation="horizontal"
+          aria-multiselectable="false"
+          className="kirk-tablist"
+        >
+          {tabs.map(tab => {
+            const isSelected = selectedTab.id === tab.id
+            return (
+              <div
+                className={cc(['kirk-tab-container', { 'kirk-tab-selected': isSelected }])}
+                key={tab.id}
+              >
+                <div className="kirk-tab-relative">
+                  <button
+                    role="tab"
+                    aria-controls={`${generateTabPanelId(tab)}`}
+                    aria-selected={isSelected ? 'true' : 'false'}
+                    title={tab.label}
+                    tabIndex={isSelected ? 0 : -1}
+                    id={tab.id}
+                    ref={this.state.tabIdToRefs.get(tab.id)}
+                    onClick={this.handleTabClicked}
+                    onKeyDown={this.handleTabKeyDown}
+                    className={cc(['kirk-tab'])}
+                  >
+                    {tab.label}
+                  </button>
+                  {tab.badgeContent && (
+                    <Badge ariaLabel={tab.badgeAriaLabel} className="kirk-tab-badge">
```

Comment:
Since the Tabs is using programmatic focus management on <button>s, this badge count will most likely never be announced for screen readers (Unless the screenreader user is using virtual cursor to access the badge content, most likely this will not happen).

We could put the badge inside the button and absolutely position it ? When the button get focused, the badge will be announced and maybe will could use aria labels to improve the announcements e.g.

<button>Messages <badge aria-label="3 new">3</badge></button> => "Messages 3 new" will be announced when button is focused.

wdyt ?

Reply:
Unfortunately we can't put the `Badge` inside the button because `<button>` does not accept block element, but what we could do is to enrich the button title attribute, and concatenate button label with `badgeAriaLabel`. What do you think?

Reply:
^ sgtm or we could hav